src/test/groovy/com/mechanitis/demo/spock/OrderingServiceSpec.groovy

```diff
@@ -0,0 +1,125 @@
+package com.mechanitis.demo.spock
+
+import com.synacy.gradprogram.spock.exercise.Cart
+import com.synacy.gradprogram.spock.exercise.Item
+import com.synacy.gradprogram.spock.exercise.ItemType
+import com.synacy.gradprogram.spock.exercise.Order
+import com.synacy.gradprogram.spock.exercise.OrderStatus
+import com.synacy.gradprogram.spock.exercise.OrderingService
+import com.synacy.gradprogram.spock.exercise.UnableToCreateOrderException
+import spock.lang.Specification
+
+class OrderingServiceSpec extends Specification {
+
+    OrderingService service
+    Cart cart
+    def items = [
+            new Item ("Rice", 50.00, ItemType.FOOD),
+            new Item ("Chicken", 20.00, ItemType.FOOD),
+            new Item ("Nasa", 34.5, ItemType.CLOTHING),
+            new Item ("Air-con", 34.5, ItemType.APPLIANCE),
+            new Item ("Laptop", 34.5, ItemType.GADGET),
+            new Item ("Nike", 500.54, ItemType.CLOTHING)
+    ]
+//    Item item1 = new Item ("Rice", 50.00, ItemType.FOOD)
+//    Item item2 = new Item ("Chicken", 20.00, ItemType.FOOD)
+//    Item item3 = new Item ("Nasa", 34.5, ItemType.CLOTHING)
+//    Item item4 = new Item ("Air-con", 34.5, ItemType.APPLIANCE)
+//    Item item5 = new Item ("Laptop", 34.5, ItemType.GADGET)
+
+
+    def setup() {
+        service = new OrderingService()
+    }
+
+    def "should return true when cart contains food item type"() {
+        when:
+        boolean containsFoodType = service.cartContainsFoodItem(cart = new Cart(UUID.randomUUID(), items))
+
+        then:
+        containsFoodType
+    }
+
+    def "should demonstrate the total cost of cart"() {
+        when:
+        double totalPrice = service.calculateTotalCostOfCart(cart = new Cart(UUID.randomUUID(), items))
+
+        then:
+        totalPrice
+    }
+
+    def "should return true when cart is eligible for discount"() {
+        given:
+        cart = new Cart(UUID.randomUUID(), items)
+
+        when:
+        boolean result = service.isCartEligibleForDiscount(cart)
+
+        then:
+        result
+    }
+
+    def "should apply discount to cart items when the cart is eligible for discount"() {
+        given:
+        cart = new Cart(UUID.randomUUID(), items)
+        List<Double> initialCosts = cart.getItems().collect() { it.getCost() }
```

Comment:
Let's try to make things as simple as we can. Let's try to avoid anything that does any kind of looping. My suggestion would be to use fewer items to make it easier.